src/pages/permissions/PermissionIdentities.tsx

```diff
@@ -0,0 +1,295 @@
+import {
+  Button,
+  Icon,
+  Row,
+  TablePagination,
+  useNotify,
+} from "@canonical/react-components";
+import { useQuery } from "@tanstack/react-query";
+import { fetchIdentities } from "api/auth-identities";
+import Loader from "components/Loader";
+import ScrollableTable from "components/ScrollableTable";
+import SelectableMainTable from "components/SelectableMainTable";
+import SelectedTableNotification from "components/SelectedTableNotification";
+import { FC, useState } from "react";
+import { useSearchParams } from "react-router-dom";
+import { queryKeys } from "util/queryKeys";
+import useSortTableData from "util/useSortTableData";
+import PermissionIdentitiesFilter, {
+  AUTH_METHOD,
+  PermissionIdentitiesFilterType,
+  QUERY,
+} from "./PermissionIdentitiesFilter";
+import { useSettings } from "context/useSettings";
+import EditIdentityGroupsBtn from "./actions/EditIdentityGroupsBtn";
+import usePanelParams, { panels } from "util/usePanelParams";
+import CustomLayout from "components/CustomLayout";
+import PageHeader from "components/PageHeader";
+import NotificationRow from "components/NotificationRow";
+import HelpLink from "components/HelpLink";
+import { useDocs } from "context/useDocs";
+import EditIdentityGroupsPanel from "./panels/EditIdentityGroupsPanel";
+import { pluralize } from "util/instanceBulkActions";
+import Tag from "components/Tag";
+
+const PermissionIdentities: FC = () => {
+  const notify = useNotify();
+  const {
+    data: identities = [],
+    error,
+    isLoading,
+  } = useQuery({
+    queryKey: [queryKeys.identities],
+    queryFn: () => fetchIdentities(),
+  });
+  const { data: settings } = useSettings();
+  const docBaseLink = useDocs();
+  const panelParams = usePanelParams();
+  const [searchParams] = useSearchParams();
+  const [selectedIdentityNames, setSelectedIdentityNames] = useState<string[]>(
```

Comment:
I think we need to use ids, the names are not guaranteed to be unique. I assume this is the source of the bug in the screencast below where the 1st identity is selected, then the "select all" click only selects the 2nd identity

[Screencast from 22.04.2024 11:20:11.webm](https://github.com/canonical/lxd-ui/assets/1155472/6e98cb74-d19c-4195-b033-a75fbe8ae3c5)

Reply:
The table is using identity IDs. I will change the state variable to `selectedIdentityIDs` to be more clear, I couldn't quite recreate this bug, although I did make an update yesterday for table `row.name` to use `identity.id` which should have fixed this issue, maybe you just needed to pull the latest changes if you haven't already done so?

Reply:
I can still reproduce this with latest changes. I think it only triggers when the Name and the id are different. Then on "select all" the entries where the name and the ID are not matching, the checkbox at the beginning of the row won't be checked.

Reply:
aah yes, this was already fixed on my local branch so I was confused, issue was `filteredName` prop for `SelectedTableNotification` was using identity names instead of ids, I've made sure all references here uses identity ids instead of names